### ISSUE TITLE
java.lang.StackOverflowError using requestPermissions(@NonNull Context var1, IPermissionsListener var2, String... var3)

### ISSUE THREAD

**Issue by benjaminVadon** [NewtronLabs/EasyPermissions#2]
Hello, 

testing your library in an small sample app, it fails in a java.lang.StackOverflowError using method with specific permissions declared : 
`EasyPermissions.getInstance().requestPermissions(App.getContext(), permissionListener,
                PERMISSION);`

version 2.2.0 and 2.2.1
error  :  
```
 java.lang.StackOverflowError: stack size 8MB
        at com.newtronlabs.easypermissions.EasyPermissions.requestPermissions(Unknown Source:9)
....
....
        at com.newtronlabs.easypermissions.EasyPermissions.requestPermissions(Unknown Source:9)
        at com.ondilo.rxandroidbletest.ble.Ble.ensureLocationPermission(Ble.java:158)
        at com.ondilo.rxandroidbletest.MainActivity.onCreate(MainActivity.java:45)

```
By decompiling app and looking at your obfusacted code (yes I must not look at your code, but it's for debug purpose ;) ), it appears that method :
`public void requestPermissions(@NonNull Context paramContext, IPermissionsListener paramIPermissionsListener, String... paramVarArgs) `

self call recursively instead of calling (obfuscated name)

`private void a(Context paramContext, IPermissionsListener paramIPermissionsListener, String... paramVarArgs)`

maybe a mistake in obfuscation step  or  just calling bad method is base code ;)

Cheer

**Comment by benjaminVadon** [comment]
Method retrieving from manifest works well 

`EasyPermissions.getInstance().requestPermissions(App.getContext(), permissionListener);`

**Comment by NewtronLabs** [comment]
Hi @benjaminVadon! 

Thank you for bringing this to our attention, while you are correct that decompiling Newtron Lab's source code is a direct violation of our license we do appreciate all your effort in finding the issue. We are currently looking to hire more developers. If you want to work in a team of talented individuals who love to develop enterprise-grade solutions send us your resume. We are always looking for passionate individuals. 

The issue has been fixed in version 2.2.2. 

Thank you,
Newtron Labs 
email: solutions@newtronlabs.com
